File: algorithms/1_easy/C/02176_count_equal_and_divisible_pairs_in_an_array/solution.py
class Solution:
    def countPairs(self, nums: List[int], k: int) -> int:

        nums_len = len(nums)
        count = 0

        for i in range(nums_len):
            for j in range((i + 1), nums_len):
                if nums[i] == nums[j] and (i * j) % k == 0:
                    count += 1

        return count

File: algorithms/1_easy/C/02176_count_equal_and_divisible_pairs_in_an_array/solution.cpp
class Solution {
public:
    int countPairs(vector<int>& nums, int k) {        
        const size_t nums_len{nums.size()};
        int count{0};

        for (size_t i{0}; i < nums_len; ++i) {
            for (size_t j{i + 1}; j < nums_len; ++j) {
                if (nums[i] == nums[j] && (i * j) % k == 0) {
                    count += 1;
                }
            }
        }

        return count;
    }
};
